nmigen/hdl/ast.py

```diff
@@ -417,6 +417,42 @@ def matches(self, *patterns):
         else:
             return Cat(*matches).any()
 
+    def rotate_left(self, offset):
+        """Rotate left by constant modulo 2**len(self).
+        
+        Parameters
+        ----------
+        offset : int, in
```

Comment:
Non-`Value` parameters are always inputs, since Python does not have out parameters. Only `Value` parameters can, in principle, be outputs.